RacingGame/Map/src/main/java/racing/map/MapPlugin.java

```diff
@@ -0,0 +1,118 @@
+package racing.map;
+
+import racing.common.data.Entity;
+import racing.common.data.GameData;
+import racing.common.data.GameImage;
+import racing.common.data.World;
+import racing.common.data.entityparts.PositionPart;
+import racing.common.data.entityparts.TilePart;
+import racing.common.map.MapSPI;
+import racing.common.map.Tile;
+import racing.common.data.TileType;
+import racing.common.services.IGamePluginService;
+
+/**
+ * Plugin used to control the map
+ *
+ */
+public class MapPlugin implements IGamePluginService, MapSPI {
+
+    @Override
+    public void start(GameData gameData, World world) {
+    }
+
+    @Override
+    public void stop(GameData gameData, World world) {
+        for (Entity tile : world.getEntities(Tile.class)) {
+            world.removeEntity(tile);
+        }
+    }
+
+    /**
+     * Finds the tile weight for the tile closed to the entity  
+     * @param p the entity
+     * @param world the world
+     * @return the weight of the closed tile
+     */
+    @Override
+    public double getPositionWeight(Entity p, World world) {
+        Entity closedTile = null;
+        double shortestDistance = Double.MAX_VALUE;
+        for (Entity tile : world.getEntities(Tile.class)) {
+            if (closedTile == null) {
+                closedTile = tile;
+                shortestDistance = distance(p, tile);
+                continue;
+            }
+
+            double nextDistance = distance(p, tile);
+            if (nextDistance < shortestDistance) {
+                closedTile = tile;
+                shortestDistance = nextDistance;
+            }
+        }
+        TilePart tp = closedTile.getPart(TilePart.class);
+        return tp.getType().getWeight();
+    }
+
+    /**
+     * Calculates the distance between to entities based center of the image
+     * @param t1 entity one
+     * @param t2 entity two
+     * @return the distance between the centers of the two entities
+     */
+    public double distance(Entity t1, Entity t2) {
+        PositionPart p1 = t1.getPart(PositionPart.class);
+        PositionPart p2 = t2.getPart(PositionPart.class);
+        
+        GameImage i1 = t1.getImage();
+        GameImage i2 = t2.getImage();
+        
+        double x = Math.pow((p2.getX() + (i2.getWidth() / 2)) - (p1.getX() + (i1.getWidth() / 2)), 2);
+        double y = Math.pow((p2.getY() + (i2.getHeight()/ 2)) - (p1.getY() + (i1.getHeight()/ 2)), 2);
+        return Math.sqrt(x + y);
+    }
+
+    /**
+     * Creates a new map based on the input arguments
+     *
+     * @param d a two dimensional array with integers representing the TileType
+     * @param gameData the GameData
+     * @param world the World
+     */
+    @Override
+    public void createMap(int[][] d, GameData gameData, World world) {
```

Comment:
As this doesn't really create the map, could the functionlity perhaps be rearranged, so that the SPI only contains one createMap method, and this is merely a private helper-function for the createMap() method? So we encapsule as much implementation as possible.

- shoot me down if there's a reason it's arranged the way it is